### ISSUE TITLE
Ctrl+Backspace not working properly in "Search in chat's history" after Select All

### ISSUE THREAD

**Issue by 21mtd** [Chatterino/chatterino2#4461]
### Checklist

- [X] I'm reporting a problem with Chatterino
- [X] I've verified that I'm running **the most** recent nightly build or stable release
- [X] I've looked for my problem on the [wiki](https://wiki.chatterino.com/Help/)
- [X] I've searched the [issues and pull requests](https://github.com/Chatterino/chatterino2/issues?q=) for similar looking reports

### Describe your issue

Instead of delete, it removes Select All highlight and move the text cursor indicator all the way back to the left.

Reproduction steps:

- Open a random chat
- Press Ctrl+F to search for history
- Type in random words `asd`
- Press Ctrl+A or Double Click input box to Select All then Ctrl+Backspace

It works normally:

- in chat.
- Select All then Backspace without Ctrl
- Ctrl+Backspace without Select All

### Screenshots

https://user-images.githubusercontent.com/75530836/226168196-4e4b4f9f-ceb3-4194-b8f9-b6b0dafbf26f.mp4

### OS and Chatterino Version

Chatterino 2.4.2 (commit c8204ef7) built with Qt 5.15.2, Windows SDK, MSVC 193431942, Crashpad Running on Windows 10 Version 2009, kernel: 10.0.19045

**Comment by SotirisVas** [comment]
From what I have tested, if you select all and ctrl+backspace, it deletes everything except the last word. Not entirely sure if this is the intention, but this is why in your video, "asd" remains, because its technically the last word. If you type multiple words and try it, you will remain with the last word typed.

**Comment by 21mtd** [comment]
I tested again, it does delete everything except the last word

**Comment by 2547techno** [comment]
Probably an issue with the underlying `QLineEdit` widget. There might be a way to make a hacky fix by hooking into its signals but I don't think that's the right way to go

**Comment by vchuynh** [comment]
I investigated this quite a bit.  I believe that the reason why selecting all the text by holding and dragging left mouse click behaves as desired is because of the cursor position.  When you use the Ctrl+A shortcut for select all, the cursor position is at the end and using Ctrl+Backspace after would produce the bug.  When you select with the mouse from left to right where the cursor is at the end again, it also produces the bug.  However, if you select right to left, it works as desired.  I tried messing with the cursor and selection related functions for QLineEdit, but I couldn't get a selection with the cursor position at the beginning.  Adding an event filter was the only way that I could fix it using QLineEdit.
